modules/light-clients/07-tendermint/types/update_test.go

```diff
@@ -282,64 +281,65 @@ func (suite *TendermintTestSuite) TestCheckHeaderAndUpdateState() {
+	suite.Require().True(ok)
+	ctx := path.EndpointA.Chain.GetContext()
+	clientStore := path.EndpointA.Chain.App.GetIBCKeeper().ClientKeeper.ClientStore(ctx, path.EndpointA.ClientID)
+	expectedProcessTime, ok := types.GetProcessedTime(clientStore, pruneHeight)
+	suite.Require().True(ok)
+	// check iteration key metadata is pruned
+	expectedConsKey := types.GetIterationKey(clientStore, pruneHeight)
+	suite.Require().NotNil(expectedConsKey)
+
+	// Increment the time by a week
+	suite.coordinator.IncrementTimeBy(7 * 24 * time.Hour)
+
+	// create the consensus state that can be used as trusted height for next update
+	path.EndpointA.UpdateClient()
+
+	// Increment the time by another week, then update the client.
+	// This will cause the first two consensus states to become expired.
+	suite.coordinator.IncrementTimeBy(7 * 24 * time.Hour)
+	path.EndpointA.UpdateClient()
+
+	ctx = path.EndpointA.Chain.GetContext()
+	clientStore = path.EndpointA.Chain.App.GetIBCKeeper().ClientKeeper.ClientStore(ctx, path.EndpointA.ClientID)
 
 	// check that the first expired consensus state got deleted along with all associated metadata
-	consState, err := types.GetConsensusState(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(currentCtx, clientID), suite.cdc, height)
+	consState, ok := path.EndpointA.Chain.GetConsensusState(path.EndpointA.ClientID, pruneHeight)
 	suite.Require().Nil(consState, "expired consensus state not pruned")
-	suite.Require().Error(err, "getting deleted consensus state did not return error")
+	suite.Require().False(ok)
 	// check processed time metadata is pruned
-	processTime, ok := types.GetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(currentCtx, clientID), height)
+	processTime, ok := types.GetProcessedTime(clientStore, pruneHeight)
 	suite.Require().Equal(uint64(0), processTime, "processed time metadata not pruned")
 	suite.Require().False(ok)
 	// check iteration key metadata is pruned
-	consKey := types.GetIterationKey(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(currentCtx, clientID), height)
+	consKey := types.GetIterationKey(clientStore, pruneHeight)
 	suite.Require().Nil(consKey, "iteration key not pruned")
 
 	// check that second expired consensus state doesn't get deleted
 	// this ensures that there is a cap on gas cost of UpdateClient
-	consState, err = types.GetConsensusState(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(currentCtx, clientID), suite.cdc, nextHeight)
-	suite.Require().Equal(consensusState, consState, "consensus state is unexpectedly pruned")
-	suite.Require().NoError(err)
-	// check processed time metadata is not pruned
-	processTime, ok = types.GetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(currentCtx, clientID), nextHeight)
-	suite.Require().Equal(uint64(ctx.BlockTime().UnixNano()), processTime, "processed time metadata is incorrect")
+	consState, ok = path.EndpointA.Chain.GetConsensusState(path.EndpointA.ClientID, pruneHeight)
```

Comment:
looks like this should be expired height and same for below

Reply:
No there are two expired consensus states. I'm calling the one that is pruned `pruneHeight` and the one that is expired but not pruned `expiredHeight`